unitstatushandler/softwaremanager.go

```diff
@@ -71,6 +71,7 @@ type softwareUpdate struct {
 
 type softwareManager struct {
 	sync.Mutex
+	runCond *sync.Cond
```

Comment:
Consider using `WaitGroup` it doesn't require additional lock and looks simplier.

Reply:
Run instances status could called before when we run instances, so in this case will be better to use cond variable to reduce potential crashed